### PR TITLE
[ANCHOR-737] Update SEP-31 test with new KYC flow

### PR DESCRIPTION
**Pull request by JiahuiWho**
### Description

This modify  SEP-12 Service  to support new KYC flow in SEP-31. 
It also updates the SEP-31 e2e test to use the new flow.

### Context

This is to support the protocol change of SEP-31.

### Testing

- `./gradlew test`

### Documentation

N/A

### Known limitations

N/A

### CODE REVIEW

File: core/src/main/java/org/stellar/anchor/sep12/Sep12Service.java [hunk 1]
```diff
@@ -175,8 +175,11 @@ void validateGetOrPutRequest(Sep12CustomerRequestBase requestBase, Sep10Jwt toke
       throws SepException {
     if (requestBase.getTransactionId() != null) {
       try {
+        // `transactionId` should be used in conjunction with customer type `type` (sep6,
+        // sep31-sender, sep-31-receiver) to get the customer account and memo
         GetTransactionResponse txn =
             platformApiClient.getTransaction(requestBase.getTransactionId());
+        String customerType = requestBase.getType();
```
Comment (philipliu) [inline core/src/main/java/org/stellar/anchor/sep12/Sep12Service.java:182]: This is not used anywhere?

File: essential-tests/src/testFixtures/kotlin/org/stellar/anchor/platform/e2etest/Sep31End2EndTests.kt [hunk 2]
```diff
@@ -148,16 +147,20 @@ open class Sep31End2EndTests : AbstractIntegrationTests(TestConfig()) {
       transfer.sign(keypair)
       wallet.stellar().submitTransaction(transfer)
     }
+    waitStatus(postTxResponse.id, SepTransactionStatus.PENDING_CUSTOMER_INFO_UPDATE)
+
+    anchor.customer(token).add(customerKycInfo)
```
Comment (philipliu) [inline essential-tests/src/testFixtures/kotlin/org/stellar/anchor/platform/e2etest/Sep31End2EndTests.kt:152]: Can we make a SEP-12 request to fetch the required fields like the SEP-6 tests?

File: core/src/main/java/org/stellar/anchor/sep12/Sep12Service.java [hunk 3]
```diff
@@ -175,8 +175,11 @@ void validateGetOrPutRequest(Sep12CustomerRequestBase requestBase, Sep10Jwt toke
       throws SepException {
     if (requestBase.getTransactionId() != null) {
       try {
+        // `transactionId` should be used in conjunction with customer type `type` (sep6,
```
Comment (philipliu) [inline core/src/main/java/org/stellar/anchor/sep12/Sep12Service.java:178]: Do we need to add a check for this?

File: kotlin-reference-server/src/main/kotlin/org/stellar/reference/event/processor/Sep31EventProcessor.kt [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+    val requiredKyc =
+      listOf("bank_account_number", "bank_account_type", "bank_number", "bank_branch_number")
+  }
+
+  override suspend fun onQuoteCreated(event: SendEventRequest) {
+    TODO("Not yet implemented")
+  }
+
+  override suspend fun onTransactionCreated(event: SendEventRequest) {
+    log.warn { "Sep31EventProcessor onTransactionCreated" }
```
Comment (philipliu) [inline kotlin-reference-server/src/main/kotlin/org/stellar/reference/event/processor/Sep31EventProcessor.kt:27]: Nit: I don't think we need the Sep31EventProcessor` prefix as the logger should handle this.

Comment (JiahuiWho) [reply kotlin-reference-server/src/main/kotlin/org/stellar/reference/event/processor/Sep31EventProcessor.kt:27]: Will remove this ... it was for print log purpose

File: core/src/main/java/org/stellar/anchor/sep12/Sep12Service.java (hunk 3 above)
Comment (philipliu) [reply core/src/main/java/org/stellar/anchor/sep12/Sep12Service.java:178]: Yeah, let's keep it optional.

**Review by philipliu — APPROVED** [review]

**Pull request merged**